node/cli/src/command.rs

```diff
@@ -409,7 +409,7 @@ pub fn run() -> Result<()> {
 				#[cfg(feature = "moonriver-native")]
 				chain_spec if chain_spec.is_moonriver() => {
 					let block: service::moonriver_runtime::Block =
-						generate_genesis_block(&chain_spec, state_version)?;
+						generate_genesis_block(&*chain_spec, state_version)?;
```

Comment:
wait, you're referencing a dereferenced pointer? couldn't you just plug in chain_spec then since chain_spec seems like it's just a pointer?

Reply:
ahh, this is from before we chatted about just looking at the parachain staking. although i'm still curious if you know why

Reply:
this is the type of stuff I will need to figure out when I update our repo to .26